src/backend/gp_libpq_fe/fe-connect.c

```diff
@@ -436,7 +436,7 @@ PQconnectStartParams(const char **keywords,
 	{
 		conn->status = CONNECTION_BAD;
 		/* errorMessage is already set */
-		return false;
+		return conn;
```

Comment:
I found gpdb has already fixed this line of code: https://github.com/xunzhang/incubator-hawq/blob/47361ae71f8c90930c68985245aeef669437574b/src/interfaces/libpq/fe-connect.c#L449.